src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -203,31 +205,89 @@ public function getDefaultPageLength()
      */
     public function addCustomPageLengthToPageLengthMenu()
     {
-        // If the default Page Length isn't in the menu's values, Add it the beginnin and resort all to show a croissant list.
-        // assume both arrays are the same length.
-        if (! in_array($this->getDefaultPageLength(), $this->getOperationSetting('pageLengthMenu')[0])) {
-            // Loop through 2 arrays of prop. pageLengthMenu
-            foreach ($this->getOperationSetting('pageLengthMenu') as $key => &$page_length_choices) {
-                // This is a condition that should be always true.
-                if (is_array($page_length_choices)) {
-                    array_unshift($page_length_choices, $this->getDefaultPageLength());
-                }
+        $values = $this->getOperationSetting('pageLengthMenu')[0];
+        $labels = $this->getOperationSetting('pageLengthMenu')[1];
+        // this is a condition that should be always true.
+        if (is_array($values) && is_array($labels)) {
+            $position = array_search($this->getDefaultPageLength(), $values);
+            // if position is not false we already have that value in the pagination array
+            // we are just going to make it first element in array
+            if ($position !== false) {
+                array_unshift($values, $this->getDefaultPageLength());
+                array_unshift($labels, $labels[$position]);
+            } else {
+                // if it's not in array we add it as the first element
+                array_unshift($values, $this->getDefaultPageLength());
+                array_unshift($labels, $this->getDefaultPageLength());
```

Comment:
@pxpm can you try this?
With my tests it solves the problem of positioning the default page length on the array.

```suggestion
                for ($i = 0; $i < count($values); $i++) {
                    if ($values[$i] > $this->getDefaultPageLength() || $values[$i] === -1) {
                        array_splice($values, $i, 0, $this->getDefaultPageLength());
                        array_splice($labels, $i, 0, $this->getDefaultPageLength());
                        break;
                    }
                    if ($i === count($values) - 1) {
                        $values[] = $this->getDefaultPageLength();
                        $labels[] = $this->getDefaultPageLength();
                        break;
                    }
                }
```

![image](https://user-images.githubusercontent.com/1838187/103368241-fe8fa300-4abe-11eb-8b29-8c16988dc3f4.png)

![image](https://user-images.githubusercontent.com/1838187/103368277-1109dc80-4abf-11eb-9a9d-d6d538c2d394.png)

Reply:
As far as I understood, it is properly selecting the default page length on the dropdown menu. I just did the part of inserting 
Maybe I'm missing something? 😅

Anyway, everything is working 👌
Tested with all the possible formats.

Reply:
You didn't. 

Everything works as expected. 

Just pushed your fix along with fixed tests for it. 

Thanks again \o